modules/rest/src/main/java/org/apache/ignite/internal/rest/RestComponent.java

```diff
@@ -115,9 +122,12 @@ public void start() {
      * @return {@code True} if server was started successfully, {@code False} if couldn't bind one of the ports.
      */
     private synchronized boolean startServer(int httpPortCandidate, int httpsPortCandidate, boolean sslEnabled, boolean dualProtocol) {
+        // Workaround to avoid micronaut race condition on startup.
+        SHARED_STARTUP_LOCK.lock();
         try {
             httpPort = httpPortCandidate;
```

Comment:
Not related to this PR, but both `httpPort` and `httpsPort` must be made `volatile`, there's a race condition in `httpPort()` and `httpsPort()`  methods